joblib/pool.py

```diff
@@ -288,14 +281,9 @@ def __init__(self, writer, reducers=None, protocol=HIGHEST_PROTOCOL):
         Pickler.__init__(self, writer, protocol=protocol)
         if reducers is None:
             reducers = {}
-        if hasattr(Pickler, 'dispatch'):
-            # Make the dispatch registry an instance level attribute instead of
-            # a reference to the class dictionary under Python 2
-            self.dispatch = Pickler.dispatch.copy()
-        else:
-            # Under Python 3 initialize the dispatch table with a copy of the
-            # default registry
-            self.dispatch_table = copyreg.dispatch_table.copy()
+        # Make the dispatch registry an instance level attribute instead of
+        # a reference to the class dictionary under Python 2
```

Comment:
As dill seems to not require a python version dependent dispatch registry, "under Python 2" should be removed.